GPU/Common/VertexDecoderArm64.cpp

```diff
@@ -0,0 +1,779 @@
+	/*
+	{&VertexDecoder::Step_PosS8Through, &VertexDecoderJitCache::Jit_PosS8Through},
+	*/
+	{&VertexDecoder::Step_PosS16Through, &VertexDecoderJitCache::Jit_PosS16Through},
+	/*
+	{&VertexDecoder::Step_PosFloatThrough, &VertexDecoderJitCache::Jit_PosFloat},
+	*/
+	{&VertexDecoder::Step_PosS8, &VertexDecoderJitCache::Jit_PosS8},
+	{&VertexDecoder::Step_PosS16, &VertexDecoderJitCache::Jit_PosS16},
+	{&VertexDecoder::Step_PosFloat, &VertexDecoderJitCache::Jit_PosFloat},
+
+	{&VertexDecoder::Step_PosS8Skin, &VertexDecoderJitCache::Jit_PosS8Skin},
+	{&VertexDecoder::Step_PosS16Skin, &VertexDecoderJitCache::Jit_PosS16Skin},
+	{&VertexDecoder::Step_PosFloatSkin, &VertexDecoderJitCache::Jit_PosFloatSkin},
+
+	/*
+	{&VertexDecoder::Step_NormalS8Morph, &VertexDecoderJitCache::Jit_NormalS8Morph},
+	{&VertexDecoder::Step_NormalS16Morph, &VertexDecoderJitCache::Jit_NormalS16Morph},
+	{&VertexDecoder::Step_NormalFloatMorph, &VertexDecoderJitCache::Jit_NormalFloatMorph},
+
+	{&VertexDecoder::Step_PosS8Morph, &VertexDecoderJitCache::Jit_PosS8Morph},
+	{&VertexDecoder::Step_PosS16Morph, &VertexDecoderJitCache::Jit_PosS16Morph},
+	{&VertexDecoder::Step_PosFloatMorph, &VertexDecoderJitCache::Jit_PosFloatMorph},
+
+	{&VertexDecoder::Step_Color8888Morph, &VertexDecoderJitCache::Jit_Color8888Morph},
+	{&VertexDecoder::Step_Color4444Morph, &VertexDecoderJitCache::Jit_Color4444Morph},
+	{&VertexDecoder::Step_Color565Morph, &VertexDecoderJitCache::Jit_Color565Morph},
+	{&VertexDecoder::Step_Color5551Morph, &VertexDecoderJitCache::Jit_Color5551Morph},
+	*/
+};
+
+
+JittedVertexDecoder VertexDecoderJitCache::Compile(const VertexDecoder &dec) {
+	dec_ = &dec;
+
+	const u32 ALL_CALLEE_SAVED = 0x7FF80000;
+	BitSet32 regs_to_save(ALL_CALLEE_SAVED);
+
+	const u8 *start = AlignCode16();
+
+	// TODO: Also push D8-D15, the fp registers we need to save.
+
+	bool prescaleStep = false;
+	bool skinning = false;
+
+	bool log = false;
+
+	// Look for prescaled texcoord steps
+	for (int i = 0; i < dec.numSteps_; i++) {
+		if (dec.steps_[i] == &VertexDecoder::Step_TcU8Prescale ||
+			dec.steps_[i] == &VertexDecoder::Step_TcU16Prescale ||
+			dec.steps_[i] == &VertexDecoder::Step_TcFloatPrescale) {
+			prescaleStep = true;
+		}
+		if (dec.steps_[i] == &VertexDecoder::Step_WeightsU8Skin ||
+			dec.steps_[i] == &VertexDecoder::Step_WeightsU16Skin ||
+			dec.steps_[i] == &VertexDecoder::Step_WeightsFloatSkin) {
+			skinning = true;
+		}
+	}
+
+	// if (skinning) log = true;
+
+	ABI_PushRegisters(regs_to_save);
+
+	// Keep the scale/offset in a few fp registers if we need it.
+	if (prescaleStep) {
+		MOVP2R(X3, &gstate_c.uv);
+		if (cpu_info.bNEON) {
+			fp.LDR(64, INDEX_UNSIGNED, neonUVScaleReg, X3, 0);
+			fp.LDR(64, INDEX_UNSIGNED, neonUVOffsetReg, X3, 8);
+			if ((dec.VertexType() & GE_VTYPE_TC_MASK) == GE_VTYPE_TC_8BIT) {
+				fp.MOVI2FDUP(neonScratchRegD, by128, scratchReg);
+				fp.FMUL(32, neonUVScaleReg, neonUVScaleReg, neonScratchRegD);
```

Comment:
I suppose if this used SCONV it would get rid of the constants entirely?

-[Unknown]